src/ray/rpc/worker/core_worker_client.h

```diff
@@ -279,6 +281,9 @@ class CoreWorkerClient : public std::enable_shared_from_this<CoreWorkerClient>,
   /// The max sequence number we have processed responses for.
   int64_t max_finished_seq_no_ GUARDED_BY(mutex_) = -1;
 
+  /// An offset subtracted from every sequence number before the request is sent.
+  uint64_t sequence_nr_offset_;
```

Comment:
Could we call this `starting_sequence_number`?

Reply:
Also, we should make sure to subtract this from client_processed_up_to as well (as well as the sequence_number) in the request.

Reply:
Good point about `client_processed_up_to`. I don't mind the name change